File: cpp/1615. 最大网络秩.cpp
class Solution {
public:
    int maximalNetworkRank(int n, vector<vector<int>>& roads) {
        vector<vector<int> > grid(n, vector<int>(n));
        vector<int> cnt(n);
        for(int i=0;i<roads.size();i++){
            int x=roads[i][0];
            int y=roads[i][1];
            grid[x][y]=1;
            grid[y][x]=1;
            cnt[x]++;
            cnt[y]++;
        }
        
        int res=0;
        for(int i=0;i<n;i++){
            for(int j=i+1;j<n;j++){
                res=max(res,cnt[i]+cnt[j]-(grid[i][j]==1?1:0));
            }
        }         
        return res;
    }
};

File: 刷题/golang/1615. 最大网络秩.go
func maximalNetworkRank(n int, roads [][]int) int {
    grid := make([][]bool, n)
    for i := range grid {
        grid[i] = make([]bool, n)
    }
    cnt := make([]int, n)
    for _, v := range roads {
        grid[v[0]][v[1]] = true
        grid[v[1]][v[0]] = true
        cnt[v[0]]++
        cnt[v[1]]++
    }
    res := 0
    for i := 0; i<n; i++ {
        for j := i+1; j<n; j++ {
            t := cnt[i]+cnt[j]
            if grid[i][j] {
                t-=1
            }
            res=max(res,t)
        }
    }
    return res;
}
func max(x, y int) int {
    if x>y {
        return x
    }
    return y
}